packages/andromeda_protocol/src/vault.rs

```diff
@@ -0,0 +1,210 @@
+use common::{
+    ado_base::{recipient::Recipient, AndromedaMsg, AndromedaQuery},
+    error::ContractError,
+    withdraw::Withdrawal,
+};
+use cosmwasm_std::{to_binary, wasm_execute, Coin, CosmosMsg, ReplyOn, Storage, SubMsg, Uint128};
+use cw_storage_plus::Map;
+use schemars::JsonSchema;
+use serde::{Deserialize, Serialize};
+
+/// Mapping between (Address, Funds Denom) and the amount
+pub const BALANCES: Map<(String, String), Uint128> = Map::new("balances");
+pub const STRATEGY_CONTRACT_ADDRESSES: Map<String, String> =
+    Map::new("strategy_contract_addresses");
+
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, JsonSchema)]
+pub enum StrategyType {
+    Anchor,
+    NoStrategy,
```

Comment:
What is the purpose of the `NoStrategy` variant? From what I can see `StrategyType` is always given as `Option<StrategyType>` which handles the `None` case.